src/utils/minichef.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  ChainId,
+  MINICHEF_CONTRACT_ADDRESSES,
+  getMinichefPid,
+} from "../constants"
+import { MulticallContract, MulticallProvider } from "../types/ethcall"
+
+import { BigNumber } from "@ethersproject/bignumber"
+import { Contract } from "ethcall"
+import MINICHEF_CONTRACT_ABI from "../constants/abis/miniChef.json"
+import { MiniChef } from "../../types/ethers-contracts/MiniChef"
+
+type MinichefData = {
+  [poolAddress: string]: { sdlPerDay: BigNumber; pid: number }
+}
+
+/**
+ * Returns sdlPerDay and pid from minichef for the given pool addresses.
+ */
+export async function getMinichefRewardsData(
+  ethCallProvider: MulticallProvider,
+  chainId: ChainId,
+  poolAddresses: string[],
+): Promise<MinichefData | null> {
+  const minichefAddress = MINICHEF_CONTRACT_ADDRESSES[chainId]
+  const addressesPidTuples = poolAddresses
+    .map((poolAddress) => [poolAddress, getMinichefPid(chainId, poolAddress)])
+    .filter(([, pid]) => !!pid) as [string, number][]
+  if (!addressesPidTuples.length || !minichefAddress || !ethCallProvider)
+    return null
+  try {
+    const minichefContract = new Contract(
+      minichefAddress,
+      MINICHEF_CONTRACT_ABI,
+    ) as MulticallContract<MiniChef>
+
+    const [saddlePerSecond, totalAllocPoint] = await ethCallProvider.tryEach(
+      [minichefContract.saddlePerSecond(), minichefContract.totalAllocPoint()],
+      [false, false],
+    )
+
+    const poolInfos = await ethCallProvider.tryEach(
+      addressesPidTuples.map(([, pid]) => minichefContract.poolInfo(pid)),
+      Array(addressesPidTuples.length).fill(true),
```

Comment:
I was actually trying to remember this style of filling arrays when I was refactoring multicalls to `tryEach`

Reply:
I feel like `calls.map(() => true)` is maybe shorter, but it all does the same thing